### PR TITLE
Created a makefile to build and test agent

### PR DESCRIPTION
**Pull request by AntonZelenin**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+SLEEP = 60
+DOCKER_COMPOSE_DEV = docker-compose-dev.yml
+DOCKER_TEST = docker exec -i anodot-agent pytest
+
+all: build test-all
+
+build:
+	docker-compose up -d --build
+	sleep $(SLEEP)
+	./setup_tests.sh
```
Comment (dariakharlan) [inline Makefile:10]: Execute setup_tests.sh under the testing command

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+test-all:
+	$(DOCKER_TEST) -x
+
+all-dev: build-dev test-all-dev
+
+build-dev:
+	docker-compose -f $(DOCKER_COMPOSE_DEV) up -d --build
+	docker exec -i anodot-agent python setup.py install
+	sleep $(SLEEP)
+	./setup_tests.sh
```
Comment (dariakharlan) [inline Makefile:21]: Execute setup_tests.sh under the testing command

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+test-all:
+	$(DOCKER_TEST) -x
+
+all-dev: build-dev test-all-dev
+
+build-dev:
+	docker-compose -f $(DOCKER_COMPOSE_DEV) up -d --build
+	docker exec -i anodot-agent python setup.py install
```
Comment (dariakharlan) [inline Makefile:19]: for the dev image you need to execute `python setup.py develop` instead of `python setup.py install`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	sleep $(SLEEP)
+	./setup_tests.sh
+
+test-all-dev:
+	$(DOCKER_TEST) -x
```
Comment (dariakharlan) [inline Makefile:24]: Before executing tests in the dev environment you need to clean up the state. Add `docker-compose down -v` and `docker-compose up -d`. For separate source tests in the `up` command specify only services that needed to execute the test. For example, for kafka you need agent, dc, Kafka and zookeeper, for elastic you need agent, dc and es.

Comment (dariakharlan) [inline Makefile:24]: add `-vv` option for testing in the dev env (separate source tests also)

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+test-directory: test-destination
+	$(DOCKER_TEST) tests/test_directory_http.py
```
Comment (dariakharlan) [inline Makefile:27]: add -x option for every separate source test

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_tcp_http.py
+
+test-destination: prepare-source nap
+	$(DOCKER_TEST_DEV) tests/test_destination.py
+
+##--------------------
+## DEPENDENCY TARGETS
+##--------------------
+build-all: get-streamsets-stages
+	docker-compose up -d --build
```
Comment (dariakharlan) [inline Makefile:51]: For a production build, we need to specify `GIT_SHA1` argument
`docker-compose build --build-arg GIT_SHA1=$(git describe --dirty --always) && docker-compose up -d`

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+	docker-compose up -d mongo
+
+build-mysql:
+	docker-compose up -d mysql
+
+build-postgres:
+	docker-compose up -d postgres
+
+get-streamsets-stages:
+	curl -L https://github.com/anodot/anodot-sdc-stage/releases/download/v1.0/anodot-1.0.tar.gz -o /tmp/sdc.tar.gz && tar xvfz /tmp/sdc.tar.gz -C streamsets/lib
```
Comment (dariakharlan) [inline Makefile:102]: please use new url https://github.com/anodot/anodot-sdc-stage/releases/download/v1.0.1/anodot-1.0.1.tar.gz
test workflow still uses the old URL, forgot to update

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	sleep $(SLEEP)
+
+build-influx:
+	docker-compose up -d influx
+
+build-kafka: build-zookeeper
+	docker-compose up -d kafka
+	sleep $(SLEEP)
+
+build-zookeeper:
```
Comment (kiamran) [inline Makefile:90]: I think there is no reason to start zookeeper without Kafka?

**Pull request merged**